src/rgw/rgw_kms.cc

```diff
@@ -688,24 +688,24 @@ class TransitSecretEngine: public VaultSecretEngine {
     int res = send_request(dpp, "POST", "", config_path,
                            post_data, y, dummy_bl);
     if (res < 0) {
-      return res;
-    }
-    if (dummy_bl.length() != 0) {
       ldpp_dout(dpp, 0) << "ERROR: unexpected response from Vault marking key to delete: "
-	<< dummy_bl
-	<< dendl;
-      return -EINVAL;
+        << dummy_bl.c_str()
+        << dendl;
+      return res;
     }
 
     res = send_request(dpp, "DELETE", "", delete_path,
                        string{}, y, dummy_bl);
     if (res < 0) {
+      ldpp_dout(dpp, 0) << "ERROR: unexpected response from Vault deleting key: "
```

Comment:
will dummy_bl contain anything if res < 0 ?  Might be more useful to log res.

Reply:
I believe it does as it might fail with an error message that can help to fix.